### PR TITLE
Turn SearchType variable into an int

### PR DESCRIPTION
**Pull request by Kiloutre**
not that it matters much but heh

### CODE REVIEW

File: controllers/search/search.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ func SearchHandler(c *gin.Context) {
 	if c.Query("order") == "true" {
 		searchForm.SortOrder = true
 	}
+	searchForm.SortType, err = strconv.Atoi(c.Query("sort"))
```
Comment (akuma06) [inline controllers/search/search.go:67]: Move this to https://github.com/NyaaPantsu/nyaa/pull/1371/files#diff-2ff0b4807eadb9a31e1e7b187f5da802R50
It's better to have it directly set in the NewSearchForm function

**Review by akuma06 — CHANGES_REQUESTED** [review]

File: controllers/search/search.go (hunk 1 above)
Comment (Kiloutre) [reply controllers/search/search.go:67]: order & sort are not related to the search results nor form but only to the refine form, more precisely only to show the correct value in the two \<select\> after a page load which is just a display thing, so putting it right under the refine bool which is also just a display thing (display refine form yes/no) seems to make sense to me

Comment (akuma06) [reply controllers/search/search.go:67]: It's not that, NewSearchForm have an argument with c *gin.Context. This function should provide a default SearchForm based on the values passed by c.
Except for refine which I agree should be set in SearchHandler, I would like order&sortType to be set in NewSearchForm. The reason is that I would like to reuse the SearchForm for things beside the Search pages (for example, in mod panel for listing pages).
Therefore, in order to not copy/paste multiple times the lines for "order" and "SortType", having them directly in NewSearchForm is better.

Comment (Kiloutre) [reply controllers/search/search.go:67]: alright but i already tried making the changes you requested and i miserably failed at it so i can't really do it anyway

**Pull request closed without merging**